src/main/scala/Tiebreaks.scala

```diff
@@ -0,0 +1,112 @@
+package strategygames.tiebreaks
+
+sealed trait Participant
+case class Player(id: String) extends Participant
+case object Virtual           extends Participant
+
+sealed trait Presence
+case object Absent  extends Presence
+case object Present extends Presence
+
+sealed trait POV
+case class Hero(participant: Player, presence: Presence) extends POV
+case class Foe(participant: Participant)                 extends POV
+
+// Results are always from a particualar point-of-view. So we only need
+// the three cases.
+sealed trait Outcome
+case object WonAgainst extends Outcome
+case object DrewWith   extends Outcome
+case object LostTo     extends Outcome
+
+case class Result(round: Int, player: Hero, outcome: Outcome, opponent: Foe)
+
+trait Tournament {
+  val nbRounds: Int
+  def resultsForPlayer(player: Player): List[Result]
+}
+
+class Tiebreak(val tournament: Tournament) {
+
+  val Win: Double  = 2
+  val Draw: Double = 1
+  val Loss: Double = 0
+  // NOTE: we use zero-base indexing here.
+
+  // In the following we distinguish between score and tiebreak.
+  // Score is what you get for the Result directly.
+  // Tiebreak is what you get from your opponent for a tiebreak
+  def scoreAtRound(p: Player, round: Int): Double =
+    tournament.resultsForPlayer(p).take(round).map(score).sum
+
+  def score(p: Player): Double =
+    scoreAtRound(p, tournament.nbRounds)
+
+  def virtualOpponentScoreAtRound(p: Player, round: Int): Double =
+    scoreAtRound(p, round) + remainingRoundDraws(round)
+
+  def remainingRoundDraws(round: Int): Double =
+    ((tournament.nbRounds) - round - 1) * Draw
+
+  def score(g: Result): Double =
+    g match {
+      case Result(_, _, WonAgainst, _) => Win
+      case Result(_, _, DrewWith, _)   => Draw
+      case Result(_, _, LostTo, _)     => Loss
+
+      // Not sure any of the rest of these make any sense.
+      case _ => Loss
+    }
+
+  // When used for tie-breaking among players with the same raw score, no
+  // multiplying is necessary and the sum of the raw scores of the opponents
+  // played is used to break ties (Golombek 1977). When used as a tie-break
+  // system, it is equivalent to the Solkoff system.
+  // - https://en.wikipedia.org/wiki/Buchholz_system
+  def buccholz(p: Player): Double =
+    tournament.resultsForPlayer(p).map(buccholzForGame).sum
+
+  def buccholzForGame(g: Result): Double =
+    g match {
+      // Regular game played with regular result
+      case Result(_, Hero(Player(_), Present), _, Foe(p: Player)) => score(p)
```

Comment:
This doesn't take into account playing against a player who had a bye (should get half point for tiebreak not full), see actual scores below in test case.

Reply:
also doesn't take into account player withdrawing and you scoring points from them (half per round missed). I think you need to extend score(p) to take this and above comment into account?